rocAL/rocAL_pybind/amd/rocal/plugin/tf.py

```diff
@@ -65,97 +60,96 @@ def __next__(self):
                 self.label_list.append(self.label_2d_numpy)
                 self.bbox_list.append(self.bb_2d_numpy)
                 sum_count = sum_count +count
-
             self.target = self.bbox_list
             self.target1 = self.label_list
             max_cols = max([len(row) for batch in self.target for row in batch])
-            # max_rows = max([len(batch) for batch in self.target])
             max_rows = 100
             bb_padded = [batch + [[0] * (max_cols)] * (max_rows - len(batch)) for batch in self.target]
             bb_padded_1=[row + [0] * (max_cols - len(row)) for batch in bb_padded for row in batch]
             arr = np.asarray(bb_padded_1)
             self.res = np.reshape(arr, (-1, max_rows, max_cols))
             max_cols = max([len(row) for batch in self.target1 for row in batch])
-            # max_rows = max([len(batch) for batch in self.target1])
             max_rows = 100
             lab_padded = [batch + [[0] * (max_cols)] * (max_rows - len(batch)) for batch in self.target1]
             lab_padded_1=[row + [0] * (max_cols - len(row)) for batch in lab_padded for row in batch]
             labarr = np.asarray(lab_padded_1)
+            label_list= np.array(lab_list)
             self.l = np.reshape(labarr, (-1, max_rows, max_cols))
-            self.num_bboxes_arr = np.array(self.num_bboxes_list)
-
             if self.tensor_dtype == types.FLOAT:
-                return self.out.astype(np.float32), self.res, self.l, self.num_bboxes_arr
+                return self.out.astype(np.float32), self.res, self.l,label_list
             elif self.tensor_dtype == types.FLOAT16:
-                return self.out.astype(np.float16), self.res, self.l, self.num_bboxes_arr
-        elif (self.loader._name == "TFRecordReaderClassification"):
-            if(self.loader._oneHotEncoding == True):
-                self.labels = np.zeros((self.bs)*(self.loader._numOfClasses),dtype = "int32")
-                self.loader.GetOneHotEncodedLabels(self.labels)
-                self.labels = np.reshape(self.labels, (-1, self.bs, self.loader._numOfClasses))
-            else:
-                self.labels = np.zeros((self.bs),dtype = "int32")
-                self.loader.GetImageLabels(self.labels)
+                return self.out.astype(np.float16), self.res, self.l,label_list
+            else: 
+                return self.out.astype(np.uint8), self.res, self.l,label_list
 
-            if self.tensor_dtype == types.FLOAT:
-                return self.out.astype(np.float32), self.labels
-            elif self.tensor_dtype == types.TensorDataType.FLOAT16:
-                return self.out.astype(np.float16), self.labels
-        
+        elif (self.loader._name == "TFRecordReaderClassification"):
+            print("CLASSIFICATION ITERATOR")
+            print(self.output_tensor_list)
+            self.output_tensor_list[0].copy_data_numpy(self.out)
```

Comment:
I dont see a if / else conditions to differentiate the data types like the condition added in tf detection.
Please add this too